maven-plugin/src/main/java/org/openmrs/maven/plugins/Run.java

```diff
@@ -73,13 +81,113 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
 			}
 		}
 
+		try {
+			File tempFolder = createTempReactorProject(server);
+			cleanInstallModule(tempFolder.getPath());
+			deleteTempReactorProject(tempFolder);
+			deployWatchedModules(server);
+		} catch (MavenInvocationException e) {
+			throw new MojoFailureException(e.getMessage());
```

Comment:
You should always try to preserve a stacktrace. Use new MojoFailureException("Failed to build watched modules", e);